src/Sentry/SentrySdk.cs

```diff
@@ -28,13 +28,39 @@ public static class SentrySdk
         /// </summary>
         public static SentryId LastEventId { [DebuggerStepThrough] get => _hub.LastEventId; }
 
+        internal static IHub InitHub(SentryOptions options)
+        {
+            // Side-effects in a factory function 🤮
+            options.SetupLogging();
+
+            // If DSN is null (i.e. not explicitly disabled, just unset), then
+            // try to resolve the value from environment.
+            var dsn = options.Dsn ??= DsnLocator.FindDsnStringOrDisable();
+
+            // If it's either explicitly disabled or we couldn't resolve the DSN
+            // from anywhere else, return a disabled hub.
+            if (Dsn.IsDisabled(dsn))
+            {
+                options.DiagnosticLogger?.LogWarning(
+                    "Init was called but no DSN was provided nor located. Sentry SDK will be disabled."
+                );
+
+                return DisabledHub.Instance;
+            }
+
+            // Validate DSN for an early exception in case it's malformed
+            _ = Dsn.Parse(dsn);
```

Comment:
When to use the discard operator vs when not to?

In this PR you've removed from a code unrelated to the changed and added here

Reply:
Here the discard is used to signal that the return of an otherwise pure function is ignored and it's only called for the exception (which is debatably a side-effect, but it's usually not considered as one). Basically, in 99.9% of cases you will want to have the return value, but this is that 0.01% case where we don't. The discard is meant to highlight that the return value (which is the most element part of this function) is ignored on purpose and is not a mistake (albeit a hack).

In other cases, like in case of `o.UseStackTraceFactory` the function is inherently effectful and it's called for its side-effects rather than the return value. Conversely to the other case, the return of `UseStackTraceFactory` is only ever needed in 0.01% of cases and then only for convenience. Here, ignoring the return value is absolutely normal.

So, in short:

Use discard to highlight that a return method of an (almost) pure function is discarded on purpose.

Don't use discard when ignoring return of an inherently effectful method, because the main point of executing said method is in the side-effects and not the return value.

Reply:
>  The discard is meant to highlight that the return value (which is the most important element of this function) is ignored on purpose and is not a mistake (albeit a hack).

And that's why I enabled warning when a return type isn't assigned or explicitly discarded.

> Don't use discard when ignoring return of an inherently effectful method, because the main point of executing said method is in the side-effects and not the return value.

I'm fine with that but please make the change in 1 PR to avoid the noise of this refactor in every single change